esphome/components/lightwaverf/LwRx.cpp

```diff
@@ -0,0 +1,427 @@
+              if (i < 0) {
+                args->rx_repeatcount++;
+              } else {
+                args->rx_repeatcount = 1;
+              }
+            }
+          } else {
+            args->rx_repeatcount = 0;
+          }
+          args->rx_prevpkttime = curr_millis;
+          // If last message hasn't been read it gets overwritten
+          memcpy(args->rx_msg, args->rx_buf, RX_MSGLEN);
+          if (args->rx_repeats == 0 || args->rx_repeatcount == args->rx_repeats) {
+            if (args->rx_pairtimeout != 0) {
+              if ((curr_millis - args->rx_pairstarttime) / 100 <= args->rx_pairtimeout) {
+                if (args->rx_msg[3] == RX_CMD_ON) {
+                  args->rx_addpairfrommsg_();
+                } else if (args->rx_msg[3] == RX_CMD_OFF) {
+                  args->rx_remove_pair_(&args->rx_msg[2]);
+                }
+              }
+            }
+            if (args->rx_report_message_()) {
+              args->rx_msgcomplete = true;
+            }
+            args->rx_pairtimeout = 0;
+          }
+          // And cycle round for next one
+          args->rx_state = RX_STATE_IDLE;
+        } else {
+          args->rx_state = RX_STATE_BYTESTARTFOUND;
+        }
+      }
+      break;
+  }
+}
+
+/**
+  Test if a message has arrived
+**/
+bool LwRx::lwrx_message() { return (rx_msgcomplete); }
+
+/**
+  Set translate mode
+**/
+void LwRx::lwrx_settranslate(bool rxtranslate) { rx_translate = rxtranslate; }
+/**
+  Transfer a message to user buffer
+**/
+bool LwRx::lwrx_getmessage(uint8_t *buf, uint8_t len) {
+  bool ret = true;
+  int16_t j = 0;  // int
+  if (rx_msgcomplete && len <= RX_MSGLEN) {
+    for (uint8_t i = 0; ret && i < RX_MSGLEN; i++) {
+      if (rx_translate || (len != RX_MSGLEN)) {
+        j = this->rx_find_nibble_(rx_msg[i]);
+        if (j < 0)
+          ret = false;
+      } else {
+        j = rx_msg[i];
+      }
+      switch (len) {
+        case 4:
+          if (i == 9)
+            buf[2] = j;
+          if (i == 2)
+            buf[3] = j;
+        case 2:
+          if (i == 3)
+            buf[0] = j;
+          if (i == 0)
+            buf[1] = j << 4;
+          if (i == 1)
+            buf[1] += j;
+          break;
+        case 10:
+          buf[i] = j;
+          break;
+      }
+    }
+    rx_msgcomplete = false;
+  } else {
+    ret = false;
+  }
+  return ret;
+}
+
+/**
+  Return time in milliseconds since last packet received
+**/
+uint32_t LwRx::lwrx_packetinterval() { return millis() - rx_prevpkttime; }
+
+/**
+  Set up repeat filtering of received messages
+**/
+void LwRx::lwrx_setfilter(uint8_t repeats, uint8_t timeout) {
+  rx_repeats = repeats;
+  rx_timeout = timeout;
+}
+
+/**
+  Add a pair to filter received messages
+  pairdata is device,dummy,5*addr,room
+  pairdata is held in translated form to make comparisons quicker
+**/
+uint8_t LwRx::lwrx_addpair(const uint8_t *pairdata) {
+  if (rx_paircount < RX_MAXPAIRS) {
```

Comment:
```suggestion
  if (this->rx_paircount < RX_MAXPAIRS) {
```

It seems you've got a bunch of places throughout where you're referring to a member variable without the `this->` prefix. We tend to prefer this is appended to each use of class variables within the code base. Please fix this for all member variable uses throughout. 😇